### PR TITLE
Bump TGUI dependencies, updates tooling

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do

- Bumps TGUI dependencies. The big one being removal of prettierx 0.19, which has been abandoned for over 2 years now, in favor of prettier 3.0.3
- Runs prettier in write mode over TGUI, bringing it into stylistic compliance (and causing major mass conflicts)
- Adds tgui/bin/tgui-formatting.bat to make it easier for initial prettier conflicts to be resolved
- Bakes in VSC build options for DM and TGUI (these might mess with user settings if already set? unsure)
- Bumps CI to node 18 from node 12
- tgui dev server is made to use legacy openssl as I can't find where to include a define for the digest function so we don't get the `ERR_OSSL_EVP_UNSUPPORTED` error. By consequence we also need `--no-experimental-fetch` or further things break.

I want to stress that no functionality changes have been made to TGUI code in this PR. It's all stylistic.
EDIT: I liked there are like 2 non-stylistic changes where things inside of maps are given keys so they are compliant. No real actual changes to end user from this though.

## Why It's Good For The Game
Updated dependencies patches security issues. If we are staying with TGUI 3 for the foreseeable future, we should look after it.
We also have like, a ton of major vulnerability alerts due to current TGUI deps. No thank you.

## Testing
Loaded a bunch of TGUIs, everything looked normal. Dev server works fine still.
**Should still TM it as I don't 100% trust automated tools.**

## Changelog
NPFC

### CODE REVIEW

File: _build_dependencies.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # For dreamchecker
 export SPACEMANDMM_TAG=suite-1.7.1
 # For TGUI
-export NODE_VERSION=12
+export NODE_VERSION=16
```
Comment (warriorstar-orion) [inline _build_dependencies.sh:5]: Node 16 stopped receiving security updates 4 days ago. Can we bump this to the next LTS release, 18?

Comment (S34NW) [reply _build_dependencies.sh:5]: sure

File: tgui/packages/tgui/webpack.config.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ module.exports = (env = {}, argv) => {
           : path.resolve(__dirname, './public/.tmp'),
       filename: '[name].bundle.js',
       chunkFilename: '[name].chunk.js',
+      hashFunction: 'SHA256',
```
Comment (Contrabang) [inline tgui/packages/tgui/webpack.config.js:32]: I think this is the default, so I think this should be fine, just want to mark it down.

File: .vscode/tasks.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"type": "dreammaker",
+			"dme": "paradise.dme",
+			"problemMatcher": [
+				"$dreammaker"
```
Comment (Contrabang) [inline .vscode/tasks.json:8]: I wasnt able to get this file to work for me, but it should probably be fine, especially for new devs setting up their workspace.

**Review by Contrabang — CHANGES_REQUESTED** [review]
Can you please update the PR description to include the Node 18 change?

File: tgui/package.json [hunk 4]
```diff
@@ -7,17 +7,18 @@
   ],
   "scripts": {
     "build": "eslint packages && cd packages/tgui && npx webpack --mode=production",
-    "watch": "cd packages/tgui-dev-server && node --experimental-modules index.js",
+    "watch": "cd packages/tgui-dev-server && set NODE_OPTIONS=--openssl-legacy-provider --no-experimental-fetch && node --experimental-modules index.js",
```
[the remark this answers is not in the document]
Comment (S34NW) [reply tgui/package.json:10]: sure

File: tgui/packages/tgui/webpack.config.js (hunk 2 above)
Comment (S34NW) [reply tgui/packages/tgui/webpack.config.js:32]: you'd think so, but without this it fails to run due to an unsupported digest (hash function)

File: tgui/packages/tgui/interfaces/BodyScanner.js [hunk 5]
```diff
@@ -199,7 +199,7 @@ const BodyScannerMainAbnormalities = (props) => {
       {abnormalities.map((a, i) => {
         if (occupant[a[0]]) {
           return (
-            <Box color={a[1]} bold={a[1] === 'bad'}>
+            <Box key={a[2]} color={a[1]} bold={a[1] === 'bad'}>
```
[the remark this answers is not in the document]
Comment (S34NW) [reply tgui/packages/tgui/interfaces/BodyScanner.js:202]: oh god right no i forgot about this, there are 2 places where map content had no key, and they require a key. The key is usually the content or whatever.

File: tgui/packages/tgui/interfaces/ERTManager.js [hunk 6]
```diff
@@ -122,165 +121,177 @@ const SendERT = (props, context) => {
...
             icon={data.secborg ? 'toggle-on' : 'toggle-off'}
             color={data.secborg ? 'red' : ''}
-            content={data.secborg ? 'Enabled' : (data.ert_type !== 'Red' ? 'Unavailable' : 'Disabled')}
+            content={
+              data.secborg
+                ? 'Enabled'
+                : data.ert_type !== 'Red'
+                ? 'Unavailable'
+                : 'Disabled'
+            }
```
[the remark this answers is not in the document]
Comment (S34NW) [reply tgui/packages/tgui/interfaces/ERTManager.js:228]: We can look at configuring the prettier rules if you'd like to try and keep some of these things. I'm not well versed in setting rules for prettier though.

File: tgui/packages/tgui/interfaces/ERTManager.js [hunk 7]
```diff
@@ -61,37 +66,31 @@ export const ERTManager = (props, context) => {
 
 export const ERTOverview = (props, context) => {
   const { act, data } = useBackend(context);
-  const {
-    security_level_color,
-    str_security_level,
-    ert_request_answered,
-  } = data;
+  const { security_level_color, str_security_level, ert_request_answered } =
+    data;
```
[the remark this answers is not in the document]
Comment (S34NW) [reply tgui/packages/tgui/interfaces/ERTManager.js:70]: As below

File: .vscode/tasks.json (hunk 3 above)
Comment (S34NW) [reply .vscode/tasks.json:8]: Strange, was it any of the build tasks or just the DM one?

Comment (Contrabang) [reply .vscode/tasks.json:8]: Ah, nevermind I got it working. Its CTRL+SHIFT+B and not an option in the launch configuration. Perhaps some documentation on that somewhere would be good

File: tgui/packages/tgui/interfaces/ERTManager.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Contrabang) [reply tgui/packages/tgui/interfaces/ERTManager.js:228]: Me neither. It works though so its fine for now. Did some research and still seems to be an open issue (one which prettierx was supposed to fix. Oh well...)

**Review by Contrabang — APPROVED** [review]
LGTM

**Review by hal9000PR — APPROVED** [review]
Fuck it we ball

**Review by lewcc — APPROVED** [review]
Haven't seen any issues, it's probably fine

**Pull request merged**